### PR TITLE
feat: add create-react-app template to build apps with IBM Carbon

### LINKED ISSUE #12598: Provide a Create React App template
It would be nice if a [custom Create React App template](https://create-react-app.dev/docs/custom-templates/) was available so that it's easy for someone to spin up an app with everything you need to start building with Carbon.

https://github.com/carbon-design-system/carbon/issues/12598#tasklist-block-49af797d-07de-4411-b72a-73d917372f3d

Ideas of what to include the template:
- Model the design off the carbon tutorial
- Include theme switching example as part of it
- UIShell with a basic CSS Grid layout
- Use a Stack within Grid
- Provide basic unit tests
  - Could we use this CRA template and it's tests as a way to smoke test new releases?
- Include some different examples of Popover usage

### LINKED ISSUE #13086: Create a Create React App template

### PR DESCRIPTION
**Pull request by Shankar-CodeJunkie**
Ref #12598
Closes #13086

Provided a Create React App template, so that it's easy for someone to spin up an app with everything you need to start building with Carbon.

**Changed**

- Used the content and the  design of the carbon tutorial
- Included theme switching example as part of it
- UIShell with a basic CSS Grid layout
- Use a Stack within Grid
- Provided basic unit tests for all the components. 

#### Testing / Reviewing

I tested the cra template locally using the file location of the template, and could see the app spinning up with Carbon design. 

`npx create-react-app my-app --template file:../carbon-bootstrapv1/ `

image

### CODE REVIEW

File: packages/cra-template/template/src/components/Header/Header.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import {render, fireEvent, screen, cleanup} from "@testing-library/react";
+import AppHeader from "./Header";
+//import renderer from 'react-test-renderer';
+afterEach(cleanup);
```
Comment (francinelucca) [inline packages/cra-template/template/src/components/Header/Header.test.js]:
```suggestion
import AppHeader from "./Header";
afterEach(cleanup);
```

File: packages/cra-template/template/src/content/HomePage/Homepage.test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    fireEvent.click(button);
+    expect(button).toBeTruthy()
+  });
+
+  it('Check tab switching', async() => {
+    const {getByRole} = render(<HomePage />);
+    const aboutTab = screen.getByText(/About/i);
+    fireEvent.click(aboutTab);
+    await screen.getByText(/What is Carbon?/i);
+    //swith another tab;
```
Comment (francinelucca) [inline packages/cra-template/template/src/content/HomePage/Homepage.test.js]:
```suggestion
    //switch to another tab;
```

File: packages/cra-template/template/src/content/HomePage/Homepage.test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    const designTab = screen.getByText('Design');
+    fireEvent.click(designTab);
+    await screen.getByText(/Rapidly build beautiful and accessible experiences/i);
+    //swith another tab;
```
Comment (francinelucca) [inline packages/cra-template/template/src/content/HomePage/Homepage.test.js]:
```suggestion
    //switch to another tab;
```

File: packages/cra-template/template/src/index.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+/*const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+    <React.StrictMode>
+      <ApolloProvider client={client}>
+        <App />
+      </ApolloProvider>
+
+    </React.StrictMode>
+);*/
```
Comment (francinelucca) [inline packages/cra-template/template/src/index.js]:
```suggestion
```

File: packages/cra-template/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Getting Started with Create React App
```
Comment (francinelucca) [inline packages/cra-template/README.md]: is this the correct Heading title?

File: packages/cra-template/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+This project helps to bootstrap your application with IBM Carbon Design framework. The application
+has 2 sample pages, and those can be accessed over ```/``` and ```/repopage``` respectively.
+It's easy for someone to spin up an app with everything you need to start building with Carbon.
+
```
Comment (francinelucca) [inline packages/cra-template/README.md:5]: Should this README also include info on how to spin up an application using this template?

**Review by francinelucca — CHANGES_REQUESTED** [review]
Confirmed can spin up and run an application using test script 🎉.

Just some suggestions.

@Shankar-CodeJunkie This PR seems to tackle bullet 1 of 3 in #12598  (correct me if I'm wrong), can we please switch out the `Closes` in the PR description for a reference? thanks!

**Comment by tay1orjones** [comment]
I just pushed an update to improve the readme docs and a few fields in the package.json

File: packages/cra-template/README.md (hunk 5 above)
Comment (tay1orjones) [reply packages/cra-template/README.md]: Fixed

File: packages/cra-template/README.md (hunk 6 above)
Comment (tay1orjones) [reply packages/cra-template/README.md:5]: Fixed

**Comment by tay1orjones** [comment]
I just pushed two more updates that correctly format the files, include the workspace dependency in the `yarn.lock`, and excludes the package from the global eslint requirements.

File: packages/cra-template/template/src/content/RepoPage/RepoPage.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+            rowCount={10}
+            headers={headers}
+          />
+        </Column>
+      </Grid>
+    );
+  }
+
+  if (error) {
+    //console.log('hey man')
```
Comment (francinelucca) [inline packages/cra-template/template/src/content/RepoPage/RepoPage.js]:
```suggestion
```

File: packages/cra-template/template/src/content/RepoPage/RepoPage.js [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+    return `Error! ${error.message}`;
+  }
+
+  if (data) {
+    //console.log('hey man1', data);
```
Comment (francinelucca) [inline packages/cra-template/template/src/content/RepoPage/RepoPage.js]:
```suggestion
```

**Review by francinelucca — CHANGES_REQUESTED** [review]
Minor cleanup that I missed the first time, otherwise this looks good!
Still think we should modify the description so that the #12598 doesn't autoclose @Shankar-CodeJunkie

**Review by francinelucca — APPROVED** [review]
🚀🔥

**Review by tay1orjones — APPROVED** [review]
@carbon-design-system/developers Since this is a new package, I'm not sure how lerna will treat this at the next release - it may pull through without issue, but it's possible it may not publish properly. We'll need to keep an eye on it during the next release.

**Comment by sstrubberg** [comment]
@Shankar-CodeJunkie Are you an IBMer? I would love to send you some recognition! What's your Slack handle?

**Comment by Shankar-CodeJunkie** [comment]
@sstrubberg :  Yes .. You can reach me at ```@Shankar Venkatachalam ``` in slack.

**Pull request merged**